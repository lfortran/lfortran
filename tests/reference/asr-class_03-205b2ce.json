{
    "basename": "asr-class_03-205b2ce",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/class_03.f90",
    "infile_hash": "cb7cff87d0874864a75325845434c59a6f1ba77970b3e24f433f0240",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-class_03-205b2ce.stdout",
    "stdout_hash": "5975ff83c80bb5f20d4d27bbb500279b2f4471fac20dd4987fe77e2a",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}