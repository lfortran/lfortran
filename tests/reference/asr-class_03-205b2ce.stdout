(TranslationUnit 
    (SymbolTable
        1
        {
            hr:
                (Module 
                    (SymbolTable
                        2
                        {
                            employee:
                                (StructType 
                                    (SymbolTable
                                        4
                                        {
                                            department:
                                                (Variable 
                                                    4 
                                                    department 
                                                    Local 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Character 1 20 () []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                ), 
                                            hire_date:
                                                (Variable 
                                                    4 
                                                    hire_date 
                                                    Local 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Integer 4 []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                ), 
                                            person:
                                                (Variable 
                                                    4 
                                                    person 
                                                    Local 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Struct 
                                                        2 person 
                                                        []
                                                    ) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                )
                                            
                                        }) 
                                    employee 
                                    [person
                                    hire_date
                                    department] 
                                    Source 
                                    Public 
                                    ()
                                ), 
                            person:
                                (StructType 
                                    (SymbolTable
                                        3
                                        {
                                            birthyear:
                                                (Variable 
                                                    3 
                                                    birthyear 
                                                    Local 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Integer 4 []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                ), 
                                            first:
                                                (Variable 
                                                    3 
                                                    first 
                                                    Local 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Character 1 20 () []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                ), 
                                            last:
                                                (Variable 
                                                    3 
                                                    last 
                                                    Local 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Character 1 20 () []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                ), 
                                            sgender:
                                                (Variable 
                                                    3 
                                                    sgender 
                                                    Local 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Character 1 1 () []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                )
                                            
                                        }) 
                                    person 
                                    [first
                                    last
                                    birthyear
                                    sgender] 
                                    Source 
                                    Public 
                                    ()
                                )
                            
                        }) 
                    hr 
                    [] 
                    .false. 
                    .false.
                ), 
            hr_code:
                (Program 
                    (SymbolTable
                        5
                        {
                            employee:
                                (ExternalSymbol 
                                    5 
                                    employee 
                                    2 employee 
                                    hr 
                                    [] 
                                    employee 
                                    Public
                                ), 
                            jack:
                                (Variable 
                                    5 
                                    jack 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Struct 
                                        5 person 
                                        []
                                    ) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            jill:
                                (Variable 
                                    5 
                                    jill 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Struct 
                                        5 employee 
                                        []
                                    ) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            person:
                                (ExternalSymbol 
                                    5 
                                    person 
                                    2 person 
                                    hr 
                                    [] 
                                    person 
                                    Public
                                )
                            
                        }) 
                    hr_code 
                    [hr] 
                    [(= 
                        (Var 5 jack) 
                        (StructTypeConstructor 
                            5 person 
                            [(StringConstant 
                                "Jack" 
                                (Character 1 4 () [])
                            )
                            (StringConstant 
                                "Smith" 
                                (Character 1 5 () [])
                            )
                            (IntegerConstant 1984 (Integer 4 []))
                            (StringConstant 
                                "M" 
                                (Character 1 1 () [])
                            )] 
                            (Struct 
                                5 person 
                                []
                            ) 
                            ()
                        ) 
                        ()
                    )
                    (= 
                        (Var 5 jill) 
                        (StructTypeConstructor 
                            5 employee 
                            [(StructTypeConstructor 
                                5 person 
                                [(StringConstant 
                                    "Jill" 
                                    (Character 1 4 () [])
                                )
                                (StringConstant 
                                    "Smith" 
                                    (Character 1 5 () [])
                                )
                                (IntegerConstant 1984 (Integer 4 []))
                                (StringConstant 
                                    "F" 
                                    (Character 1 1 () [])
                                )] 
                                (Struct 
                                    5 person 
                                    []
                                ) 
                                ()
                            )
                            (IntegerConstant 2003 (Integer 4 []))
                            (StringConstant 
                                "sales" 
                                (Character 1 5 () [])
                            )] 
                            (Struct 
                                5 employee 
                                []
                            ) 
                            ()
                        ) 
                        ()
                    )
                    (Print 
                        () 
                        [(StructInstanceMember 
                            (Var 5 jack) 
                            3 first 
                            (Character 1 20 () []) 
                            ()
                        )
                        (StructInstanceMember 
                            (Var 5 jack) 
                            3 last 
                            (Character 1 20 () []) 
                            ()
                        )
                        (StructInstanceMember 
                            (Var 5 jack) 
                            3 birthyear 
                            (Integer 4 []) 
                            ()
                        )
                        (StructInstanceMember 
                            (Var 5 jack) 
                            3 sgender 
                            (Character 1 1 () []) 
                            ()
                        )] 
                        () 
                        ()
                    )
                    (Print 
                        () 
                        [(StructInstanceMember 
                            (StructInstanceMember 
                                (Var 5 jill) 
                                4 person 
                                (Struct 
                                    5 person 
                                    []
                                ) 
                                ()
                            ) 
                            3 first 
                            (Character 1 20 () []) 
                            ()
                        )
                        (StructInstanceMember 
                            (StructInstanceMember 
                                (Var 5 jill) 
                                4 person 
                                (Struct 
                                    5 person 
                                    []
                                ) 
                                ()
                            ) 
                            3 last 
                            (Character 1 20 () []) 
                            ()
                        )
                        (StructInstanceMember 
                            (StructInstanceMember 
                                (Var 5 jill) 
                                4 person 
                                (Struct 
                                    5 person 
                                    []
                                ) 
                                ()
                            ) 
                            3 birthyear 
                            (Integer 4 []) 
                            ()
                        )
                        (StructInstanceMember 
                            (StructInstanceMember 
                                (Var 5 jill) 
                                4 person 
                                (Struct 
                                    5 person 
                                    []
                                ) 
                                ()
                            ) 
                            3 sgender 
                            (Character 1 1 () []) 
                            ()
                        )
                        (StructInstanceMember 
                            (Var 5 jill) 
                            4 department 
                            (Character 1 20 () []) 
                            ()
                        )
                        (StructInstanceMember 
                            (Var 5 jill) 
                            4 hire_date 
                            (Integer 4 []) 
                            ()
                        )] 
                        () 
                        ()
                    )]
                )
            
        }) 
    []
)
